pkg/core/native/crypto_test.go

```diff
@@ -29,6 +29,40 @@ func TestSha256(t *testing.T) {
 		require.Equal(t, "47dc540c94ceb704a23875c11273e16bb0b8a87aed84de911f2133568115f254", hex.EncodeToString(c.sha256(ic, []stackitem.Item{stackitem.NewByteArray([]byte{1, 0})}).Value().([]byte)))
 	})
 }
+func TestKeccak256_Compat(t *testing.T) {
+	c := newCrypto()
+	ic := &interop.Context{VM: vm.New()}
+
+	testCases := []struct {
+		name         string
+		input        []byte
+		expectedHash string
+		expectPanic  bool
+	}{
+		{"bad arg type", nil, "", true},
+		{"good", []byte{1, 0}, "628bf3596747d233f1e6533345700066bf458fa48daedaf04a7be6c392902476", false},
+		{"hello world", []byte("Hello, World!"), "acaf3289d7b601cbd114fb36c4d29c85bbfd5e133f14cb355c3fd8d99367964f", false},
+		{"keccak", []byte("Keccak"), "868c016b666c7d3698636ee1bd023f3f065621514ab61bf26f062c175fdbe7f2", false},
+		{"cryptography", []byte("Cryptography"), "53d49d225dd2cfe77d8c5e2112bcc9efe77bea1c7aa5e5ede5798a36e99e2d29", false},
+		{"testing123", []byte("Testing123"), "3f82db7b16b0818a1c6b2c6152e265f682d5ebcf497c9aad776ad38bc39cb6ca", false},
+		{"long string", []byte("This is a longer string for Keccak256 testing purposes."), "24115e5c2359f85f6840b42acd2f7ea47bc239583e576d766fa173bf711bdd2f", false},
+		{"blank string", []byte(""), "c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470", false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.expectPanic {
+				require.Panics(t, func() {
+					_ = c.keccak256(ic, []stackitem.Item{stackitem.NewInterop(nil)})
```

Comment:
But why interop?

Reply:
If this test structure doesn't suit for failing test cases handling, then a separate test for failing test cases can be added below.

Reply:
See TestStdLibItoaAtoi as an example.